packages/auth/src/clients/service.test.ts

```diff
@@ -0,0 +1,332 @@
+          }
+        })
+
+        expect(validClient).toEqual(false)
+      })
+
+      test('Cannot validate client with incorrect uri', async (): Promise<void> => {
+        const expDate = new Date()
+        expDate.setTime(expDate.getTime() + 1000 * 60 * 60)
+        const nbfDate = new Date()
+        nbfDate.setTime(nbfDate.getTime() - 1000 * 60 * 60)
+        nock(KEY_REGISTRY_ORIGIN)
+          .get(TEST_KID_PATH)
+          .reply(200, {
+            display: {
+              name: 'Test Client',
+              uri: 'https://example.com/wrong'
+            },
+            keys: {
+              ...TEST_CLIENT_KEY,
+              exp: Math.round(expDate.getTime() / 1000),
+              nbf: Math.round(nbfDate.getTime() / 1000),
+              revoked: false
+            }
+          })
+
+        const validClient = await clientService.validateClientWithRegistry({
+          display: TEST_CLIENT_DISPLAY,
+          key: {
+            proof: 'httpsig',
+            jwk: TEST_CLIENT_KEY
+          }
+        })
+
+        expect(validClient).toEqual(false)
+      })
+    })
+
+    test('Cannot validate client with key that has invalid properties', async (): Promise<void> => {
+      const futureDate = new Date()
+      futureDate.setTime(futureDate.getTime() + 1000 * 60 * 60)
+      const nbfDate = new Date()
+      nbfDate.setTime(nbfDate.getTime() - 1000 * 60 * 60)
+      nock(KEY_REGISTRY_ORIGIN)
+        .get(TEST_KID_PATH)
+        .reply(200, {
+          display: TEST_CLIENT_DISPLAY,
+          keys: {
+            ...TEST_CLIENT_KEY,
+            exp: Math.round(futureDate.getTime() / 1000),
+            nbf: Math.round(nbfDate.getTime() / 1000),
+            revoked: false
+          }
+        })
+
+      nock(KEY_REGISTRY_ORIGIN).get('/wrong').reply(200)
+
+      // Validate kid
+      const validClientKid = await clientService.validateClientWithRegistry({
+        display: TEST_CLIENT_DISPLAY,
+        key: {
+          proof: 'httpsig',
+          jwk: {
+            ...TEST_CLIENT_KEY,
+            kid: 'https://openpayments.network/wrong'
+          }
+        }
+      })
+
+      expect(validClientKid).toEqual(false)
+
+      // Validate "kty"
+      const validClientKty = await clientService.validateClientWithRegistry({
+        display: TEST_CLIENT_DISPLAY,
+        key: {
+          proof: 'httpsig',
+          jwk: {
+            ...TEST_CLIENT_KEY,
+            kty: 'EC'
+          }
+        }
+      })
+
+      expect(validClientKty).toEqual(false)
+
+      // Validate "key_ops"
+      const validClientKeyOps = await clientService.validateClientWithRegistry({
+        display: TEST_CLIENT_DISPLAY,
+        key: {
+          proof: 'httpsig',
+          jwk: {
+            ...TEST_CLIENT_KEY,
+            key_ops: ['wrapKey']
+          }
+        }
+      })
+
+      expect(validClientKeyOps).toEqual(false)
+
+      // Validate "x"
+      const validClientX = await clientService.validateClientWithRegistry({
```

Comment:
You could move this (and maybe the `kid` check?) to its own test, and remove any nock from this test, since I think validation fails before the request is sent.